src/Models/ModelInstantiator.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace PDGA\DataObjects\Models;
+
+use PDGA\DataObjects\Attributes\Column;
+
+use ReflectionProperty;
+
+class ModelInstantiator
+{
+    /**
+     * Converts an associate array to a validated instance of a data object class.
+     *
+     * @param array  $arr   an array of values corresponding to the public properties of the data object.
+     * @param string $class Fully-qualified class name of the data object.
+     *
+     * @return object
+     */
+    public function arrayToDataObject(
+        array  $arr,
+        string $class
+    ): object
+    {
+        $instance = new $class();
+
+        // TODO validate.
+
+        // Assign all public properties.
+        foreach (array_keys(get_class_vars($class)) as $property)
+        {
+            if (!isset($arr[$property]))
+            {
+                continue;
+            }
+
+            $instance->{$property} = $arr[$property];
+        }
+
+        return $instance;
+    }
+
+    /**
+     * Converts a data object to an associative array with keys matching database fields for the
+     * corresponding database model.
+     *
+     * @param object $data_object An instance of a hydrated data object.
+     *
+     * @throws \ReflectionException
+     *
+     * @return array
+     */
+    public function dataObjectToDatabaseModel(
+        object $data_object
+    ): array
+    {
+        $model_array = [];
+
+        // Loop through all assigned properties of the object.
+        foreach (get_object_vars($data_object) as $property => $value)
+        {
+            $property_reflection = new ReflectionProperty($data_object::class, $property);
+
+            // Get the Column attribute.
+            $attribute = $property_reflection->getAttributes(Column::class);
```

Comment:
I'm not sure if there will ever be a case where we call this method with a Data Object that has a property that doesn't map to a Column.  If so, this `$attribute` array would be empty and the next line would crash.  Can the next line be skipped for properties that are not Column-attributed?